### PR TITLE
Adding `why` data for reported posts.

### PR DESCRIPTION
**Pull request by BrockA**
**Warning: untested!**

Per [Braiam's feature request](http://chat.meta.stackexchange.com/transcript/message/4166606#4166606), The `!!/why` command will now tell who manually reported a post and in what forum.

### CODE REVIEW

File: chatcommunicate.py [hunk 1]
```diff
@@ -98,14 +99,14 @@ def watcher(ev, wrap2):
...
-        r = handle_commands(content_source.lower(), message_parts, ev_room, ev_user_id, ev_user_name, wrap2, content_source, message_id)
+        r = handle_commands(content_source.lower(), message_parts, ev_room, ev_room_name, ev_user_id, ev_user_name, wrap2, content_source, message_id)
         if r is not None:
             message_with_reply = ":{} {}".format(message_id, r)
             if len(message_with_reply) <= 500 or "\n" in r:
                 ev.message.reply(r, False)
 
 
-def handle_commands(content_lower, message_parts, ev_room, ev_user_id, ev_user_name, wrap2, content, message_id):
+def handle_commands(content_lower, message_parts, ev_room, ev_room_name, ev_user_id, ev_user_name, wrap2, content, message_id):
```
Comment (Undo1) [inline chatcommunicate.py]: This breaks at least one other spot where we call this method:

```
TypeError: handle_commands() takes exactly 9 arguments (8 given)
2015-11-09 00:11:57.355260 UTC
  File "/Users/smokey/Stack Exchange/SmokeDetector/excepthook.py", line 46, in run_with_except_hook
    run_old(*args, **kw)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)

  File "/Users/smokey/Stack Exchange/SmokeDetector/ChatExchange/chatexchange/browser.py", line 696, in _runner
    self.on_activity(json.loads(a))

  File "/Users/smokey/Stack Exchange/SmokeDetector/ChatExchange/chatexchange/rooms.py", line 81, in on_activity
    event_callback(event, self._client)

  File "/Users/smokey/Stack Exchange/SmokeDetector/chatcommunicate.py", line 86, in watcher
    result = handle_commands(current_message.lower(), current_message.split(" "), ev_room, ev_user_id, ev_user_name, wrap2, current_message, message_id)
```

**Comment by BrockA** [comment]
Darn it; I thought i did a global search. :(

**Comment by BrockA** [comment]
I did.  GitHub's web search only shows the "top 2" matches.  Grrrr.

**Comment by BrockA** [comment]
@Undo1, Sorry about that.  [GitHub's web search only showed 2 hits for `handle_commands`](https://github.com/Charcoal-SE/SmokeDetector/search?utf8=%E2%9C%93&q=handle_commands) and I didn't think to double check that. :(

If I do much more of this, I'll recreate a proper Python dev environment on my local machine.

**Pull request merged**